exercises/book-store/canonical-data.json

```diff
@@ -0,0 +1,106 @@
+{
+   "book-store": {
+      "description": [
+                       "Calculate lowest price for shopping basket only ",
+                       "containing books from a single series.  There is no ",
+                       "discount advantage for having more than one copy of ",
+                       "any single book in a grouping."
+                     ],
+      "cases": [
+         {
+            "description": "A basket containing only a single book.",
+            "basket": [1],
+            "targetgrouping": [[1]],
+            "expected": 8.0
+         },
+         {
+            "description":  "A basket containing only two of the same book.",
+            "basket": [2, 2],
+            "targetgrouping": [[2], [2]],
+            "expected": 16.0
+         },
+         {
+            "description": "No charge to carry around an empty basket.",
+            "basket": [],
+            "targetgrouping": [[]],
```

Comment:
I actually think `[]` is OK here (and better than `[[]]`), since `[]` still is of the type "array of arrays of integers", it just happens to be an *empty* array of arrays of integers (it could be an empty array of anything,, really). Why do I think `[]` is better than `[[]]`? Well, we don't say `[[2], [2], []]` right? So we never have empty inner arrays. Thus, `[]` rather than `[[]]`.